website/project/model.py

```diff
@@ -331,55 +333,64 @@ def create(cls, auth, **kwargs):
         return comment
 
     def edit(self, content, auth, save=False):
+        log_dict = {
+                    'project': self.node.parent_id,
+                    'node': self.node._id,
+                    'user': self.user._id,
+                    'comment': self._id,
+                    }
+        if isinstance(self.root_target, StoredFileNode):
+            log_dict['file'] = {'name': self.root_target.name, 'url': self.get_comment_page_url()}
         self.content = content
         self.modified = True
         self.node.add_log(
             NodeLog.COMMENT_UPDATED,
-            {
-                'project': self.node.parent_id,
-                'node': self.node._id,
-                'user': self.user._id,
-                'comment': self._id,
-            },
+            log_dict,
             auth=auth,
             save=False,
         )
+        self.save()
         if save:
-            self.save()
+            self.node.save()
 
     def delete(self, auth, save=False):
+        log_dict = {
+                    'project': self.node.parent_id,
+                    'node': self.node._id,
+                    'user': self.user._id,
+                    'comment': self._id,
+                    }
         self.is_deleted = True
-        if self.page == Comment.FILES:
+        if isinstance(self.root_target, StoredFileNode):
+            log_dict['file'] = {'name': self.root_target.name, 'url': self.get_comment_page_url()}
             self.node.commented_files[self.root_target._id] -= 1
             if self.node.commented_files[self.root_target._id] == 0:
                 del self.node.commented_files[self.root_target._id]
         self.node.add_log(
             NodeLog.COMMENT_REMOVED,
-            {
-                'project': self.node.parent_id,
-                'node': self.node._id,
-                'user': self.user._id,
-                'comment': self._id,
-            },
+            log_dict,
             auth=auth,
             save=False,
         )
+        self.save()
         if save:
-            self.save()
+            self.node.save()
 
     def undelete(self, auth, save=False):
         self.is_deleted = False
-        if self.page == Comment.FILES:
+        log_dict = {
+                    'project': self.node.parent_id,
+                    'node': self.node._id,
+                    'user': self.user._id,
+                    'comment': self._id,
+                    }
+        if isinstance(self.root_target, StoredFileNode):
+            log_dict['file'] = {'name': self.root_target.name, 'url': self.get_comment_page_url()}
             file_key = self.root_target._id
             self.node.commented_files[file_key] = self.node.commented_files.get(file_key, 0) + 1
         self.node.add_log(
```

Comment:
I don't think this log ever gets saved. Can you add `self.node.save()` after the log?